### ISSUE TITLE
Formatting with pug, the prettier-ignore causes the code to be deleted

### ISSUE THREAD

**Issue by richex-cn** [vuejs/vetur#3165]
- [x] I have searched through existing issues
- [x] I have read through [docs](https://vuejs.github.io/vetur)
- [x] I have read [FAQ](https://vuejs.github.io/vetur/guide/FAQ.html)
- [x] I have tried restarting VS Code or running `Vetur: Restart VLS`

## Info

- Platform: Win 21H1
- Vetur version: 0.34.1
- VS Code version: 1.60.2

## Problem

When using `//- prettier-ignore` or `// prettier-ignore` at the end of the `<template lang="pug">`, the code will be accidentally deleted.

The [@prettier/plugin-pug](https://github.com/prettier/plugin-pug) plugin also had problems formatting pug files before 1.16.7 release, but this was fixed in [1.16.7](https://github.com/prettier/plugin-pug/blob/main/CHANGELOG.md#1167)

As the [documentation says](https://github.com/vuejs/vetur/blob/v0.34.1/docs/guide/formatting.md?plain=1#L21): It'll prefer to use the local version. So I assume that when I install version 1.16.7 locally, prettier will also use this version for work.

But unfortunately I got the unexpected result of the `<template lang="pug">` for formatting vue files mentioned in [#292](https://github.com/prettier/plugin-pug/issues/292) 

There are no err message in Panel -> Output -> Vue Language Server.

## Reproducible Case

I wrote a sample see https://github.com/richex-cn/sample-pug-formatting/blob/master/src/Test.vue

**Comment by Shinigami92** [comment]
First we could try upgrade this line to `1.16.7`

https://github.com/vuejs/vetur/blob/c97734bc89431d9bbc75f129d9593193d80c1800/server/package.json#L35

**Comment by richex-cn** [comment]
@Shinigami92 Reference to the [documentation](https://github.com/vuejs/vetur/wiki/Developing-Language-Server-Features) and it works fine, but this seems to have another problem: after installing the specified version of the prettier dependency locally, Why not  prefer to use the local version?

https://github.com/richex-cn/sample-pug-formatting/blob/master/package.json#L9

**Comment by Shinigami92** [comment]
@richex-cn Did you check that the exact same example worked in raw pug files?  
I'm still not safe about the stability of this `//- prettier-ignore` directive...

But if it is really working in pug itself, it should also work in `.vue` files, cause the logic is exactly the same. It just reads the content between the `<template lang="pug">(.*)</template>` and formats it with pug. (at least that's somewhat the logic behind, in real `prettier` just provide the embedded content, same as with markdown and so on)

**Comment by richex-cn** [comment]
@Shinigami92 Yes, Version 1.16.7 has been working fine in the pug file. I'm not sure if the way I debug VLS is correct, but after I manually upgrade the @prettier/plugin-pug version in vls package.json, the vue file can also work normally:

https://github.com/vuejs/vetur/blob/v0.34.1/server/package.json#L35

So I think the problem now is how to determine that vetur successfully loaded the locally installed prettier in the 0.34.1 version of vetur, so that we can know where the problem is. After solving this problem, I can also test the 1.16.7 update extensively in my project. 🤞

**Comment by yoyo930021** [comment]
Vetur will use local prettier when `vetur.useWorkspaceDependencies` true.

**Comment by yoyo930021** [comment]
In the meantime, I will update the built-in dependencies when next version

**Comment by richex-cn** [comment]
@yoyo930021 Thank you very much, everything worked fine after I set `useWorkspaceDependencies` to true, but I have a few questions I want to know:
1. In the process of determining the problem, I also suspected the useWorkspaceDependencies option, but because it was written "[Currently only for Typescript](https://github.com/vuejs/vetur/blob/v0.34.1/package.json#L226)", I missed it. It seems that this description will produce a wrong understanding. Is there a better comment instead?
2. Why is there no `useWorkspaceDependencies` option in the workspace settings?

**Comment by yoyo930021** [comment]
> 1. In the process of determining the problem, I also suspected the useWorkspaceDependencies option, but because it was written "Currently only for Typescript", I missed it. It seems that this description will produce a wrong understanding. Is there a better comment instead?

Oh no, the option description is not updated.

> 2. Why is there no useWorkspaceDependencies option in the workspace settings?

I don't know this. It is here when I start to contribute the project.

**Comment by richex-cn** [comment]
@yoyo930021 aha, So this seems to be related to VSCode.

BTW, Maybe can mention the useWorkspaceDependencies option around [this sentence](https://github.com/vuejs/vetur/blob/v0.34.1/docs/guide/formatting.md?plain=1#L21) to prevent misunderstanding.

**Comment by yoyo930021** [comment]
> @yoyo930021 aha, So this seems to be related to VSCode.
> 
> BTW, Maybe can mention the useWorkspaceDependencies option around [this sentence](https://github.com/vuejs/vetur/blob/v0.34.1/docs/guide/formatting.md?plain=1#L21) to prevent misunderstanding.

Thank you, I update it.
